File: ReadMLB.Web.Scout/src/app/players/player.model.ts
import { TeamModel } from '../teams/team.model';

export class PlayerModel {
  public firstName: string;
  public lastName: string;
  public shirt: number;
  public primaryPosition: string;
  public secondaryPosition: string;
  public bats: string;
  public throws: string;
  public team: TeamModel;
  public teamHistory?: PlayerTeamHistoryModel[];

  public position(): string {
    let retVal = this.primaryPosition;
    if (this.secondaryPosition && this.primaryPosition !== this.secondaryPosition) {
      retVal += ' / ' + this.secondaryPosition;
    }
    return retVal;
  }
  public fullName(): string {
    return this.firstName + ' ' + this.lastName;
  }

}

export interface PlayerTeamHistoryModel {
  teamId: number;
  teamAbr: string;
  teamName: string;
  organizationId: number;
  organization: string;
  league: number;
  year: number;
  inPO: boolean;
  slot: number;
}

File: ReadMLB.Web.API/Model/PlayerModel.cs

namespace ReadMLB.Web.API.Model
{
    public class PlayerModel
    {
        public long PlayerId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public byte? Shirt { get; set; }
        public string PrimaryPosition { get; set; }
        public string SecondaryPosition { get; set; }

        public string Bats { get; set; }
        public string Throws { get; set; }
        public TeamModel Team { get; set; }
    }
}
